### ISSUE TITLE
Expose API to submit recordings feedback

### ISSUE THREAD

**Issue by orontee** [suaviloquence/mopidy-listenbrainz#5]
# Context

Listenbrainz API has few paths related to recording feedbacks (love/hate), see [Feedback API](https://listenbrainz.readthedocs.io/en/latest/users/api/recordings.html#feedback-api).

This issue to discuss whether it make sense to expose a way to submit feedbacks and collect user feedbacks.

At present time, Iris implement this on its own using Last.fm; It expects each Iris instance to configure a token.

# Proposal

* Make Mopidy-Listenbrainz exposes  `submit_feedbacks()` and `collect_user_feedbacks()` actions through Mopidy API.
* Implement the corresponding three state button in [Argos](https://github.com/orontee/argos).
* (Optional) Make Mopidy-Listenbrainz exposes loved tracks as a playlist.

Remark: It could be a better design to have Mopidy API to expose a generic feedback API and have Mopidy-Listenbrainz implement a "backend" for that API; My intuition is that it could be easily done in a second step.

@suaviloquence What do you think?

**Comment by suaviloquence** [comment]
I don't know if we want to couple extensions together like that; then frontends like argos would have to explicitly check for the presence of mp-listenbrainz, which is kind of against the mopidy philosophy.  I like the idea of having a loved track playlist, though.  What if (at least until Mopidy introduces the functionality to mark a track as loved/liked/etc.) we added functionality so that if the user adds the song to the Listenbrainz "Loved Tracks" playlist, the extension then marks that song as loved in the Listenbrainz API?

**Comment by orontee** [comment]
> I don't know if we want to couple extensions together like that; then frontends like argos would have to explicitly check for the presence of mp-listenbrainz, which is kind of against the mopidy philosophy. 

I agree. Let's put this on hold: I'll start a discussion for the love/hate API in Mopidy first, see https://discourse.mopidy.com/t/how-to-implement-recordings-feedback/5952.

> I like the idea of having a loved track playlist, though. What if (at least until Mopidy introduces the functionality to mark a track as loved/liked/etc.) we added functionality so that if the user adds the song to the Listenbrainz "Loved Tracks" playlist, the extension then marks that song as loved in the Listenbrainz API?

I'd better work on a future proof solution.

**Comment by suaviloquence** [comment]
Awesome, let me know how it goes.  If you still wanted to play around with making a (readonly?) loved tracks playlist from LB i'd definitely merge that, but no worries if not yet.

**Comment by orontee** [comment]
The subject does not seem to excite the crowds: Almost no reaction to my [message](https://discourse.mopidy.com/t/how-to-implement-recordings-feedback/5952) on Discourse!

**Comment by suaviloquence** [comment]
yeah i'm not sure how active the discourse is (at least the ideas forum).  feel free to implement something here if you'd like, but no pressure if it doesn't seem like something you want to do.
